src/Diagnostics/Diagnostics.jl

```diff
@@ -0,0 +1,372 @@
+function node_adjustment(i, j, Nq, x, xmax, y, ymax)
+    # node on the corner of domain is not repeated
+    if ((x == 0 || abs(x - xmax) <= 0.001)
+        && (ymax == 0 || abs(y - 1500) <= 0.001))
+        # cancel out the 4 times repetition below
+        bound = 4 
+    # node on the edge of the domain
+    elseif (x == 0 || abs(x - xmax) <= 0.001
+            || ymax == 0 || abs(y - 1500) <= 0.001)
+        # half its repetition
+        bound = 2
+    # node on not on any considered boundary
+    else
+        # no adjustment
+        bound = 1
+    end
+
+    if ((i == 1 || i == Nq) && (j == 1 || j == Nq))
+        # corner node repeated 4 times for horizontal considerations
+        rep = 1/4 * bound
+    elseif (i == 1 || i == Nq || j == 1 || j == Nq)
+        # edge node repeated 2 times for horizontal considerations
+        rep = 1/2 * bound
+    else
+        # inner node nothing done
+        rep = 1 * bound
+    end
+
+    return rep
+end
+
+function compute_horzsums!(FT, state, i, j, k, ijk, ev, eh, e, x, y, z,
+                           Nq, xmax, ymax, Nqk, nvertelem, localaux, κ,
+                           LWP, thermoQ, horzsums, repdvsr)
+    rep = node_adjustment(i, j, Nq, x, xmax, y, ymax)
+    th = thermo_vars(thermoQ[ijk,e])
+    hs = horzavg_vars(horzsums[k,ev])
+    hs.ρ         += rep * state.ρ
+    hs.ρu        += rep * state.ρu[1]
+    hs.ρv        += rep * state.ρu[2]
+    hs.ρw        += rep * state.ρu[3]
+    hs.e_tot     += rep * state.ρe
+    hs.ρq_tot    += rep * state.moisture.ρq_tot
+    hs.q_liq     += rep * th.q_liq
+    hs.q_vap     += rep * th.q_vap
+    hs.θ_liq_ice += rep * th.θ_liq_ice
+    hs.θ_dry     += rep * th.θ_dry
+    hs.θ_v       += rep * th.θ_v
+    hs.e_int     += rep * th.e_int
+
+    # liquid water path
+    # this condition is also going to be used to get the number of points that
+    # exist on a horizontal plane provided all planes have the same number of
+    # points
+    # TODO adjust for possibility of non equivalent horizontal slabs
+    if ev == floor(nvertelem/2) && k == floor(Nqk/2)
+        # TODO: uncomment the line below after rewriting the LWP assignment below using aux.∫dz...?
+        # aux = extract_aux(dg, localaux, ijk, e)
+        LWP[1] += rep * (localaux[ijk,1,e] + localaux[ijk,2,e]) / κ 
+        repdvsr[1] += rep # number of points to be divided by
+    end
+end
+
+function compute_diagnosticsums!(FT, state, i, j, k, ijk, ev, eh, e, x, y, z,
+                                 Nq, xmax, ymax, zvals, thermoQ, horzavgs, dsums)
+    rep = node_adjustment(i, j, x, xmax, y, ymax, Nq)
+    th = thermo_vars(thermoQ[ijk,e])
+    ha = horzavg_vars(horzavgs[k,ev])
+    ds = diagnostic_vars(dsums[k,ev])
+
+    u̅ = state.ρu[1] / state.ρ
+    v̅ = state.ρu[2] / state.ρ
+    w̅ = state.ρu[3] / state.ρ
+    q̅_tot = state.moisture.ρq_tot / state.ρ
+    ũ = ha.ρu / ha.ρ
+    ṽ = ha.ρv / ha.ρ
+    w̃ = ha.ρw / ha.ρ
+    ẽ = ha.e_tot / ha.ρ
+    q̃_tot = ha.ρq_tot / ha.ρ
+
```

Comment:
I would suggest changing the notation to avoid confusion. The tilde and overbar look too similar and can lead to coding mistakes.
For example:
mean u                       --> \underline{u} 
instantaneous u:       --> u
fluctuation of u:        --> \tilde{u}

or anything that clearly distinguishes each quantity

Reply:
I don't have a preference -- your thoughts @akshaysridhar?

Reply:
Agree with `instantaneous u: --> u` , this affects `185 to 188` and quantities dependent on these variables. Fluctuating quantities seem fine with `\prime` to me. No preference for the rest.